test/unit-tests/expression/node/FunctionAssignmentNode.test.js

```diff
@@ -302,6 +302,20 @@ describe('FunctionAssignmentNode', function () {
     }, /Callback function must return a Node/)
   })
 
+  it('should throw an error when having duplicate variables', function () {
```

Comment:
I noticed
```ts
  const name = typeof param === 'string' ? param : param.name
```
up above.

Do you think it's worth adding a test case like
```ts
assert.throws(function () {
      console.log(new FunctionAssignmentNode('f', ['x', { name: 'x' }], new ConstantNode(2)))
    }, new Error('Duplicate parameter name "x"'))
```
that uses the `{ name: ... }` version?

Reply:
That's a good idea, thanks!

Done